docs/instructionset.tex

```diff
@@ -0,0 +1,323 @@
+\documentclass{scrartcl}
+\usepackage{cmap}
+\usepackage[T1]{fontenc}
+\usepackage{lmodern}
+\usepackage[english]{babel}
+\usepackage{amsmath}
+\usepackage{booktabs, array}
+\usepackage{tikz, environ}
+\usepackage{pgffor, fp}
+\usepackage{hyperref}
+
+%%%%%%%%%
+% SETUP %
+%%%%%%%%%
+
+\usetikzlibrary{decorations.pathreplacing, calligraphy, arrows.meta}
+
+\DeclareMathOperator{\op}{op}
+\DeclareMathOperator{\instr}{instr}
+
+\makeatletter
+\newsavebox{\measure@tikzpicture}
+\NewEnviron{scalepicture}[1]{%
+  \def\tikz@width{#1}%
+  \def\tikzscale{1}\begin{lrbox}{\measure@tikzpicture}%
+  \BODY
+  \end{lrbox}%
+  \pgfmathparse{#1/\wd\measure@tikzpicture}%
+  \edef\tikzscale{\pgfmathresult}%
+  \BODY
+}
+\makeatother
+
+%%%%%%%%%%%%
+% DOCUMENT %
+%%%%%%%%%%%%
+
+\title{Instruction set reference}
+\author{%
+  Jean \textsc{Caspar},
+  Loïc \textsc{Chevalier},
+  Vladimir \textsc{Ivanov},
+  Adrien \textsc{Mathieu}
+}
+\date{}
+
+\begin{document}
+\maketitle{}
+\section*{Hardware requirements}
+This instruction set requires 16 registers, \(r_0,\ldots,r_{15}\).\par
+The first register always contains zeros. It can be used as any other register.\par
+All the other registers are general purpose 16 bit register, which can be
+accessed both for writing and for reading, even in a single cycle, in which case
+the data read is the one stored in the register \textit{before} the write
+operation. Every register must initially be initialized to zeros, as should the
```

Comment:
```suggestion
operation. At startup, every register must be reset to contain only zeros, as should the
```

Reply:
Il n'y a aucun changement avec l'ancienne version... (mon problème étant le "**initially** be **initialized**")

Reply:
oui, c'est bon aussi

Reply:
EDIT : "including the instruction pointer register" aussi je pense qu'il faut rajouter